_posts/2021-05-14-bring-gitops-to-your-functions-with-argocd-part-2-the-missing-piece-tekton.md

```diff
@@ -0,0 +1,612 @@
+
+## Introduction
+
+We talked about _How we can bring GitOps principles to management of OpenFaaS functions_ in the previous blog post. If you haven't read it yet, you can follow this [link](https://www.openfaas.com/blog/bring-gitops-to-your-openfaas-functions-with-argocd/), because it might help you to understand the all of the pieces of CI/CD pipeline we want to build here. In this guide, we'll add CI (Continious Integration) part to our pipeline by using _Tekton_. After that, to be able to trigger this pipeline based on _Github_ events, we'll use an another component of [Tekton](https://tekton.dev) called [Tekton Trigger](https://tekton.dev/docs/triggers/). We will do this demo using KinD on a kubernetes running locally. Because of we are in a private network, we have to listen events that send by Github to trigger our Tekton Pipeline. So we need to find a way to susbcribe those events, and this is where Tekton Triggers comes into the picture. So, we said that everyhing is in local, so, we should open our local services to the internet, Github in this case, to be able Github send events to our event listener, and this is where Inlets, a Cloud Native Tunnel, comes in to the picture. At the end of this tutorial, we'll have a pipeline like the following:
+
+![tekton_argocd_arch](/images/bring-gitops-to-your-functions-with-argocd-part-2-the-missing-piece-tekton/argo_cd_tekton.png)
+
+## Prerequisites
+
+We have to install the following tools to be able to achieve this demo, luckily we have [arkade](https://github.com/alexellis/arkade) which is an open-source Kubernetes marketplace, so we can install of the following tools by using arkade:
+
+ * [arkade](https://get-arkade.dev) (v0.7.15) Kubernetes marketplace
+
+  ```bash
+  # Run with or without sudo
+  $ curl -sLS https://dl.get-arkade.dev | sudo sh
+  ```
+
+* KinD (Kubernetes in Docker) v0.10.0
+
+  Kubernetes is our recommendation for teams running at scale, but in this demo we will be
+    using [KinD](https://kind.sigs.k8s.io/docs/user/quick-start/) for the sake of simplicity.
+
+  ```bash
+  $ arkade get kind --version=v0.10.0
+  ```
+
+* kubectl v1.21.0
+
+  You can control your cluster using [kubectl](https://github.com/kubernetes/kubectl) CLI.
+
+  ```bash
+  $ arkade get kubectl --version=v1.21.0
+  ```
+
+* tkn v0.18.0
+
+  You can use  [tkn](https://github.com/tektoncd/cli) CLI for interacting with Tekton resources.
+
+  ```bash
+  $ arkade get tkn --version=v0.18.0
+  ```
+
+* ArgoCD CLI v2.0.0
+
+  ArgoCD CLI controls an Argo CD server. More detailed installation instructions can be found via
+    the [CLI installation documentation](https://argoproj.github.io/argo-cd/cli_installation/). Fortunately, we can
+    install it via arkade too.
+
+  ```bash
+  $ arkade get argocd --version=v2.0.0
+  ```
+
+### Setup
+
+### 1. Provision a local Kubernetes Cluster with KinD
```

Comment:
I think we should remove everything between 1 and point 3, since it's all duplicated from the previous blog post.

Instead of repeating all the content, can you link to the previous blog post and then focus on the new parts?